dna/tests/src/project/project_init.rs

```diff
@@ -0,0 +1,121 @@
+#[cfg(test)]
+pub mod tests {
+    use crate::project::fixtures::fixtures::ProjectMetaFixturator;
+    use ::fixt::prelude::*;
+    use hdk::prelude::*;
+    use hdk_crud::WrappedAgentPubKey;
+    use holochain_types::prelude::option_entry_hashed;
+    use holochain_types::prelude::ElementFixturator;
+    use holochain_types::prelude::ValidateDataFixturator;
+    use projects::project::error::Error;
+    use projects::project::project_meta::crud::ProjectMeta;
+    use projects::project::project_meta::validate::*;
+    use projects::init;
+    use projects::project::member::entry::Member;
+    use assert_matches::assert_matches;
+
+    /// Unit testing the `init()` function that is called to ensure an agent can receive signals
+    /// and is listed as a member in the project. There are two main functions called within `init()`: `create_receive_signal_cap_grant()`
+    /// and `join_project_during_init()`. The first function creates a capabilities grant token
+    /// allowing other agents to call a specified zome function (used for sending signals for responsive UI updates)
+    /// and the second function creates a member entry and links it to MEMBER_PATH 
+    #[test]
+    fn test_init() {
+               
+        let mut mock_hdk = MockHdkT::new();
+        
+        let zome_info = fixt!(ZomeInfo);
+        // init calls create_receive_signal_cap_grant, which calls zome_info and create_cap_grant
+        mock_hdk
+            .expect_zome_info()
+            .times(1)
+            .return_const(Ok(zome_info.clone())); 
+
+        // create_cap_grant calls just `create` under the hood
+        let mut functions: GrantedFunctions = BTreeSet::new();
+        functions.insert((zome_info.zome_name, "recv_remote_signal".into()));
+        // expected is for the .with, and is b/c create parameter is of type EntryWithDefId
+        let expected = EntryWithDefId::new(
+            EntryDefId::CapGrant,
+            Entry::CapGrant(CapGrantEntry {
+                tag: "".into(),
+                access: ().into(),
+                functions,
+            }),
+        );
+        let header_hash = fixt!(HeaderHash);
+        mock_hdk
+            .expect_create()
+            .with(mockall::predicate::eq(expected))
+            .times(1)
+            .return_const(Ok(header_hash.clone()));
```

Comment:
to make this modular, and reusable in the future, let's set up a function with a signature like this (it returns the unit type `()` which we can omit which is quivalent): 
```
fn setup_create_receive_signal_cap_grant_mock(mock_hdk: MockHdkT) {
  
}
```

move the code highlighted in to there